sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -135,6 +135,38 @@ abstract class JdbcDialect extends Serializable with Logging {
     s""""$colName""""
   }
 
+  /**
+   * Get the SQL that should be used to create a table. Dialects can
+   * override this method to return a SQL that works best in a particular database.
+   * To allow certain options to append when create a new table, which can be
+   * table_options or partition_options.
+   * E.g., "CREATE TABLE t (name string) ENGINE=InnoDB DEFAULT CHARSET=utf8"
+   * @param tableName The name of the table.
+   * @param strSchema The scheme string corresponding to the dialect, the schemaString method of
+   *                  JdbcUtils has provided the function of calculating the scheme string
+   *                  for different databases
+   * @param createTableOptions The string of the options that to allow certain options to append
+   *                           when create a new table, which can be table_options
+   *                           or partition_options.
+   */
+  def createTable(
+      statement: Statement,
+      tableName: String,
+      strSchema: String,
+      options: JdbcOptionsInWrite): Unit = {
+    val createTableQuery = getCreateTableQuery(tableName, strSchema, options.createTableOptions)
```

Comment:
why do we still need `getCreateTableQuery`? dialect can override `createTable` directly.

Reply:
This is a reference to other logic, similar to `getAddColumnQuery`, dialect only needs to override syntax-related things, and does not need to care about other common logic

Reply:
`getAddColumnQuery` is used by `alterTable`. The idea is, `alterTable` has too many parts and it's too much to override `alterTable` if only the ADD COLUMN syntax needs to be adjusted.

Can we follow `createSchema` here?